crates/apub/src/fetcher/search.rs

```diff
@@ -75,142 +49,113 @@ pub async fn search_by_apub_id(
+#[derive(Debug)]
+pub enum SearchableObjects {
+  Person(Person),
+  Community(Community),
+  Post(Post),
+  Comment(Comment),
+}
 
-      let person = get_or_fetch_and_upsert_person(person_uri, context, recursion_counter).await?;
-      ROR {
-        person: blocking(context.pool(), move |conn| {
-          PersonViewSafe::read(conn, person.id)
-        })
-        .await?
-        .ok(),
-        ..ROR::default()
-      }
-    }
-    SearchAcceptedObjects::Group(g) => {
-      let community_uri = g.id(&query_url)?;
-      let community =
-        get_or_fetch_and_upsert_community(community_uri, context, recursion_counter).await?;
-      ROR {
-        community: blocking(context.pool(), move |conn| {
-          CommunityView::read(conn, community.id, None)
-        })
-        .await?
-        .ok(),
-        ..ROR::default()
-      }
-    }
-    SearchAcceptedObjects::Page(p) => {
-      let p = Post::from_apub(&p, context, &query_url, recursion_counter).await?;
-      ROR {
-        post: blocking(context.pool(), move |conn| PostView::read(conn, p.id, None))
-          .await?
-          .ok(),
-        ..ROR::default()
-      }
-    }
-    SearchAcceptedObjects::Comment(c) => {
-      let c = Comment::from_apub(&c, context, &query_url, recursion_counter).await?;
-      ROR {
-        comment: blocking(context.pool(), move |conn| {
-          CommentView::read(conn, c.id, None)
-        })
-        .await?
-        .ok(),
-        ..ROR::default()
-      }
-    }
-  })
+#[derive(Deserialize)]
+#[serde(untagged)]
+pub enum SearchableApubTypes {
+  Group(Group),
+  Person(ApubPerson),
+  Page(Page),
+  Note(Note),
 }
 
-async fn delete_object_locally(query_url: &Url, context: &LemmyContext) -> Result<(), LemmyError> {
-  let res = find_object_by_id(context, query_url.to_owned()).await?;
-  match res {
-    Object::Comment(c) => {
-      blocking(context.pool(), move |conn| {
-        Comment::update_deleted(conn, c.id, true)
-      })
-      .await??;
+impl ApubObject for SearchableObjects {
+  fn last_refreshed_at(&self) -> Option<NaiveDateTime> {
+    match self {
+      SearchableObjects::Person(p) => p.last_refreshed_at(),
+      SearchableObjects::Community(c) => c.last_refreshed_at(),
+      SearchableObjects::Post(p) => p.last_refreshed_at(),
+      SearchableObjects::Comment(c) => c.last_refreshed_at(),
     }
-    Object::Post(p) => {
-      blocking(context.pool(), move |conn| {
-        Post::update_deleted(conn, p.id, true)
-      })
-      .await??;
+  }
+
+  // TODO: this is inefficient, because if the object is not in local db, it will run 4 db queries
+  //       before finally returning an error. it would be nice if we could check all 4 tables in
+  //       a single query.
+  //       we could skip this and always return an error, but then it would not be able to mark
+  //       objects as deleted that were deleted by remote server.
```

Comment:
@dessalines Any idea about this? Its not a big problem but would be nice to fix it.

Reply:
Why not parse and split the DbUrl to determine the type. IE `.../comment/...`, `.../acitivity/...`

Reply:
Because we want to be compatible with other projects, and we dont know what their urls look like.